### DISCUSSION TITLE
Unable to find Provider for jakarta.ws.rs.client.ClientBuilder vs. jakarta.ws.rs.client.ClientBuilder: org.jboss.resteasy.client.jaxrs.internal.ResteasyClientBuilderImpl not a subtype

### DISCUSSION THREAD

**Discussion by dhladky** [resteasy/resteasy#3981] [General]
Hello wise ones, 

I am migrating an old application, that used RestEasy (3.x) with the old javax notation on EAP. However the new Keycloak switched to using jakarta.ws.rs.client so I am desperately trying to find a way how to create a compatible client. 

When I use JakartaEE 10 dependency (one of the requirements of RestEasy v. 6.x is WS RS v. 3.1, which is fulfilled), I get 
jakarta.ws.rs.client.ClientBuilder: org.jboss.resteasy.client.jaxrs.internal.ResteasyClientBuilderImpl not a subtype

error. If I use the older version, I get
java.lang.ClassNotFoundException: Provider for jakarta.ws.rs.client.ClientBuilder cannot be found

The second case is not that surprising as the WS RS is only 3.0, but I wonder if there is a way around the first case. My guess is the problem of this case is with EAP having a module with RestEasy 3.x, which collides with the v. 6.2.6  I am trying to add to the application war. Is there any way, how to run newer RestEasy on EAP and if so, do you know some way, how to make it work?

I found this old atricle https://dzone.com/articles/jboss-modules-suck-it%E2%80%99s and also an article about upgrading a WildFly modules using Galleon (which is not part of EAP 7.4).

Do you know about any tutorial/readme/example/article or whatever resource I can use? Thank you.

**Comment by jamezp** [comment]
@dhladky Are you using Keycloak on the client side? If so, that won't really work well with EAP 7. It's possible you could exclude the `jaxrs` subsystem if you don't have any end points in your WAR and only use the client, but even then I'm not sure it will work correctly.

Is there any chance you could use WildFly 30? It would work there for sure. If the App itself is old you could try using https://github.com/wildfly-extras/deployment-transformer-feature-pack which will change the `javax` namespace to `jakarta` in the bytecode during deployment time. It will be a temporary workaround at least.

**Reply by dhladky** [reply]
Yes, I use the Keycloak client in my application and switching to Wildfly would be difficult. I would need to redo the whole deployment process. 

Thank you very much for your insight. I am afraid you are correct about the possible problems, so it looks like the least bad solution would be to fail forward and bypass the outdated library endpoint and parse the Keycloak REST data directly from JSON (using the old RestEasy bundled with EAP).

**Reply by jamezp** [reply]
Ah, okay. If there is anything I can help with on helping upgrade anything let me know. I'd be happy to help.

**Comment by KUPRAM-PRAMOD** [comment]
I am still facing above issue, using Resteasy 6.2.11 and wildfly 34.0.1.Final. is there any update on this issue

**Reply by jamezp** [reply]
Are you also using Keycloak?

**Reply by KUPRAM-PRAMOD** [reply]
I have resolved this issue. FYI, not using Keycloak.
